liota/lib/transports/cipethernetIP.py

```diff
@@ -0,0 +1,111 @@
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import logging
+import os
+import sys
+import time
+import cpppo
+from random import randint
+from cpppo.server.enip import client
+from cpppo.server.enip.getattr import attribute_operations
+
+log = logging.getLogger(__name__)
+
+
+class CIPEthernetIP:
+    '''
+            EtherNet Industrial Protocol implementation for LIOTA. It uses python-cpppo internally.
+    '''
+
+    def __init__(self, host, port=None, timeout=None, dialect=None, profiler=None, udp=False, broadcast=False, source_address=None):
+        """
+        :param host: CIP EtherNet/IP IP
+        :param port: CIP EtherNet/IP Port
+        :param timeout: Connection timeout
+        :param dialect: An EtherNet/IP CIP dialect, if not logix.Logix
+        :param profiler: If using a Python profiler, provide it to disable around I/O code
+        :param udp: Establishes a UDP/IP socket to use for request (eg. List Identity)
+        :param broadcast: Avoids connecting UDP/IP sockets; may receive many replies
+        :param source_address: Bind to a specific local interface (Default: 0.0.0.0:0)
+        """
+
+        self.host = host
+        self.port = port
+        self.timeout = timeout
+        self.dialect = dialect
+        self.profiler = profiler
+        self.udp = udp
+        self.broadcast = broadcast
+        self.source_address = source_address
+
+    def connect(self):
+        with client.connector(host=self.host) as self.conn:
+            if(self.conn):
+                log.info("Connected to Server")
+
+    def send(self, tag, elements, data, tag_type):
+        self.tag = tag
+        self.elements = elements
+        self.data = data
+        self.tag_type = tag_type
+
+        try:
+            req = self.conn.write(self.tag, elements=self.elements, data=self.data,
+                                  tag_type=self.tag_type)
+        except AssertionError as exc:
+            log.info("Response timed out!!")
+        except socket.error as exc:
+            log.error("Couldn't send command: %s" % (exc))
+
+    def receive(self):
+        with self.conn:
+            try:
+                request_ = self.conn.read("Scada[0]")
```

Comment:
I think "Scada" is the name/tag of the queue. Correct me if i am wrong. It seems to be hard-coded between liota and simulator. Make this also a configurable parameter in sampleProp.conf like IP address.

Reply:
Yes, it is the tag. I will change it.